### PR TITLE
#150400838 hasVoted field to comment

### PR DESCRIPTION
**Pull request by andela-ojoloko**
#### What does this PR do?

- it includes the `hasUpvoted` field as part of the response when retrieving comments.

#### Description of Task to be completed?
- update the comments and votes model
- update endpoint to get all comments
- write tests to ensure the `hasVoted`  field is included as part of the response when retrieving comments 
#### How should this be manually tested?
- At the root of the project directory run the following command:
``` bash
  pytest -v or pytest
```

#### What are the relevant pivotal tracker stories?

#150400838

### CODE REVIEW

File: api/helpers.py [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
+from flask import g
 
 def validate_type(item, input_type):
     return type(item) is input_type
 
 def serialize_thread_comments(comment):
+    user_id=g.current_user.user_id
     thread_comment_serialized = comment.serialize()
-    thread_comment_serialized["replies"] = [reply.serialize() for reply in comment.replies]
+    thread_comment_serialized["hasUpvoted"] = comment.has_upvoted(user_id)
```
Comment (andela-oowonikoko) [inline api/helpers.py]: Nitpick! The `hasUpvoted` key can be renamed `hasVoted` instead. Let this change reflect in the your method name in the `models.py` file 🙇‍♀️

File: api/models.py [hunk 2]
```diff
@@ -98,6 +98,15 @@ class Comment(db.Model, ModelOpsMixin):
     updated_at = db.Column(db.DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
     replies = db.relationship('Comment', cascade="all,delete-orphan",
                                 backref=db.backref('parent', remote_side='Comment.id'))
+    votes = db.relationship("Vote", backref='comment',
+            cascade="all, delete-orphan", lazy='dynamic')
+
+    def has_upvoted(self, loggedin_user_id):
+        vote=Vote.query.filter_by(user_id=loggedin_user_id, comment_id=self.id).first()
+        if vote:
```
Comment (andela-oowonikoko) [inline api/models.py]: I believe line 106 - 109 can be refactored to a single line `return vote is not None`. 
Also, nitpick, the model method can be renamed  `has_voted` .

File: test/api/views/test_contributions.py [hunk 3]
```diff
@@ -886,6 +886,72 @@ def test_fetching_contribution_comments(self):
...
+
+        contribution = Contribution(
+            id = self.contribution_id,
+            title = "contribution title",
+            description = "contribution description",
+            contribution_type_id = "-KA3Q0D45T6",
+            user_id = "user1"
+        )
+        db.session.add(contribution)
+        db.session.commit()
```
Comment (andela-oowonikoko) [inline test/api/views/test_contributions.py]: Please change all the `db.session.add() db.session.commit()` to `item.save()`. We now have a `save()` method in our models that handles that.

File: api/helpers.py [hunk 4]
```diff
@@ -1,12 +1,23 @@
-from .models import Tag
+from .models import Tag, Vote
 from uuid import uuid4
+from flask import g
 
 def validate_type(item, input_type):
     return type(item) is input_type
 
+def has_voted(**kwargs):
```
Comment (georgreen) [inline api/helpers.py]: Add function doc string, to make your code more clear, in terms of what the inputs contain. i.e document **kwargs

**Comment by andela-oowonikoko** [comment]
LGTM

**Pull request merged**